x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -65,6 +64,40 @@ const resolverTreeResponse = (state: DataState): ResolverTree | undefined => {
   return state.tree?.lastResponse?.successful ? state.tree?.lastResponse.result : undefined;
 };
 
+function currentRelatedEventRequestID(state: DataState): number | undefined {
+  if (state.currentRelatedEvent) {
+    return state.currentRelatedEvent?.dataRequestID;
+  } else {
+    return undefined;
+  }
+}
+
+function currentNodeEventsInCategoryRequestID(state: DataState): number | undefined {
+  if (state.nodeEventsInCategory?.pendingRequest) {
+    return state.nodeEventsInCategory.pendingRequest?.dataRequestID;
+  } else if (state.nodeEventsInCategory) {
+    return state.nodeEventsInCategory?.dataRequestID;
+  } else {
+    return undefined;
+  }
+}
+
+export const eventsInCategoryResultIsStale = createSelector(
+  currentNodeEventsInCategoryRequestID,
+  refreshCount,
```

Comment:
Can this ever be `undefined`? Seems like we initialize with `0` in the reducer right?